storm/src/main/storm/mesos/util/ZKClient.java

```diff
@@ -0,0 +1,117 @@
+package storm.mesos.util;
+
+import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.curator.framework.api.BackgroundCallback;
+import org.apache.curator.framework.api.CuratorEvent;
+import org.apache.curator.framework.api.CuratorListener;
+import org.apache.zookeeper.data.Stat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ZKClient allows you to interact with ZooKeeper. Primarily used for tracking logviewer state on hosts thus far but
+ * can be used to track any metadata needed in the future.
+ */
+
+public class ZKClient {
+  CuratorFramework _client;
+  public static final Logger LOG = LoggerFactory.getLogger(ZKClient.class);
+  private static final int BASE_SLEEP_TIME_MS = 1000;
+  private static final int MAX_RETRIES = 3;
+
+  public ZKClient() {
+    ExponentialBackoffRetry retryPolicy = new ExponentialBackoffRetry(BASE_SLEEP_TIME_MS, MAX_RETRIES);
+    _client = CuratorFrameworkFactory.newClient("localhost:2181", retryPolicy);
+    _client.start();
+  }
+
+  public ZKClient(String connectionString) {
+    LOG.info("Attempting to connect to following ZooKeeper servers: {}", connectionString);
+    ExponentialBackoffRetry retryPolicy = new ExponentialBackoffRetry(BASE_SLEEP_TIME_MS, MAX_RETRIES);
+    _client = CuratorFrameworkFactory.newClient(connectionString, retryPolicy);
+    _client.start();
+  }
+
+  public ZKClient(String connectionString, int connectionTimeout, int sessionTimeout) {
+    LOG.info("Attempting to connect to following ZooKeeper servers: {}", connectionString);
+    ExponentialBackoffRetry retryPolicy = new ExponentialBackoffRetry(BASE_SLEEP_TIME_MS, MAX_RETRIES);
+    _client = CuratorFrameworkFactory.builder()
+                                     .connectString(connectionString)
+                                     .retryPolicy(retryPolicy)
+                                     .connectionTimeoutMs(connectionTimeout)
+                                     .sessionTimeoutMs(sessionTimeout)
+                                     .build();
+    _client.start();
+  }
+
+  public boolean createNode(String path, String data) {
+    try {
+      _client.create().creatingParentsIfNeeded().forPath(path, data.getBytes());
+      return true;
+    } catch (Exception e) {
+      LOG.error(e.toString());
+      return false;
+    }
+  }
+
+  public boolean createNode(String path) {
+    try {
+      // default payload of byte[0]
+      _client.create().creatingParentsIfNeeded().forPath(path);
```

Comment:
@erikdw A ZK Node is a combination of a path name and some type of data payload (in bytes). I was just specifying that for this createNode() method, you will not need to specify any data to stuff into the ZK Node and it will automatically default to byte[0] which is pretty much nothing

Reply:
@changreytang : I'll try to put a clarifying comment in, I still don't really get it.

Reply:
My clarifying comment: https://github.com/mesos/storm/commit/84c34faae55dfbebb2343f0d83d3a40340c08f7b